File: src/main/java/com/nhi/blogly/services/TagService.java
package com.nhi.blogly.services;

import com.nhi.blogly.domain.entities.Tag;

import java.util.List;
import java.util.Set;
import java.util.UUID;

public interface TagService {

    List<Tag> getTags();

    List<Tag> createTags(Set<String> names);

    void deleteTag(UUID id);
}

File: frontend/src/services/tagService.ts
import { apiService } from './apiService';
import { Tag } from '../types/Tag';

class TagService {
  private api = apiService.getApi();

  public async getTags(): Promise<Tag[]> {
    const response = await this.api.get<Tag[]>('/tags');
    return response.data;
  }

  public async createTags(names: string[]): Promise<Tag[]> {
    const response = await this.api.post<Tag[]>('/tags', { names });
    return response.data;
  }

  public async deleteTag(id: string): Promise<void> {
    await this.api.delete(`/tags/${id}`);
  }
}

export const tagService = new TagService();
